beacon-chain/rpc/beaconv1/blocks.go

```diff
@@ -24,20 +116,217 @@ func (bs *Server) ListBlockHeaders(ctx context.Context, req *ethpb.BlockHeadersR
 // new block into its state, and therefore validate the block internally, however blocks which fail the validation are
 // still broadcast but a different status code is returned (202).
 func (bs *Server) SubmitBlock(ctx context.Context, req *ethpb.BeaconBlockContainer) (*ptypes.Empty, error) {
-	return nil, errors.New("unimplemented")
+	blk := req.Message
+	root, err := blk.HashTreeRoot()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not tree hash block: %v", err)
+	}
```

Comment:
Looks like the root is not used until line 145. Please move this closer to when it is used so bad requests can fail faster (and cheaper).

Reply:
This is used in the defer func before the line you mentioned.